tensorflow/python/util/nest.py

```diff
@@ -142,7 +152,21 @@ def _sequence_like(instance, args):
         d[key] = result[key]
       return d
     else:
-      return instance_type((key, result[key]) for key in instance)
+      d = instance_type()
+      for key in instance:
```

Comment:
This piece is identical to the one in the if branch above it. You can factor it out:

```
if instance_type == _collections.defaultdict:
  d = _collections.defaultdict(instance.default_factory)
else:
  d = instance_type()

for key in instance:
  ...
```

Reply:
The changes have been implemented.